flake.nix

```diff
@@ -0,0 +1,191 @@
+{
+  description = "A flake for working with a bridle workspace";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+
+    workspace = {
+      url = "git+file:.?dir=nix-workspace&ref=HEAD"; # Path in this same repo
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    zephyr-sdk = {
+      type = "github";
+      owner = "Irockasingranite";
+      repo = "zephyr-sdk-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    sphinx-tsn-theme-src = {
+      type = "github";
+      owner = "tiacsys";
+      repo = "sphinx-tsn-theme";
+      ref = "2022.10.2";
+      flake = false;
+    };
+
+    sphinxcontrib-svg2pdfconverter-src = {
+      type = "github";
+      owner = "missinglinkelectronics";
+      repo = "sphinxcontrib-svg2pdfconverter";
+      ref = "v1.2.2";
+      flake = false;
+    };
+
+    sphinx-csv-filter-src = {
+      type = "github";
+      owner = "crate";
+      repo = "sphinx_csv_filter";
+      ref = "0.4.1";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, workspace, zephyr-sdk, sphinx-tsn-theme-src, sphinxcontrib-svg2pdfconverter-src, sphinx-csv-filter-src }:
+    let
+      pkgs = import nixpkgs {
+        system = "x86_64-linux";
+      };
+      zephyr-sdk-pkg = zephyr-sdk.packages.x86_64-linux.zephyr-sdk-0-16-4;
+      pypkgs = pkgs.python311Packages;
+
+      sphinx-tsn-theme = pkgs.python311Packages.buildPythonPackage {
+        pname = "sphinx-tsn-theme";
+        version = "2022.10.2";
+        format = "pyproject";
+        src = sphinx-tsn-theme-src;
+        nativeBuildInputs = [
+          pypkgs.setuptools
+        ];
+      };
+
+      sphinxcontrib-svg2pdfconverter = pkgs.python311Packages.buildPythonPackage {
+        pname = "sphinxcontrib-svg2pdfconverter";
+        version = "1.2.2";
+        format = "pyproject";
+        src = sphinxcontrib-svg2pdfconverter-src;
+        nativeBuildInputs = [
+          pypkgs.setuptools
+        ];
+      };
+
+      sphinx-csv-filter = pkgs.python311Packages.buildPythonPackage {
+        pname = "sphinx-csv-filter";
+        version = "0.4.1";
+        format = "pyproject";
+        src = sphinx-csv-filter-src;
+        nativeBuildInputs = [
+          pypkgs.setuptools
+        ];
+      };
+
+      bridle-deps = [
+        # Zephyr SDK
+        zephyr-sdk-pkg
+
+        # Required build tools
+        pkgs.git
+        pkgs.cmake
+        pkgs.dtc
+        pkgs.ninja
+
+        # Other tools
+        pkgs.openocd
+
+        # Dependencies for building docs
+        pkgs.doxygen
+        pkgs.graphviz
+        pkgs.mscgen
+
+        # Python with packages that need to be available for import
+        (pkgs.python311.withPackages (ps: with ps; [
+
+          # Python dependencies (Zephyr)
```

Comment:
**question:** if we dissolve the various requirements.txt in this way, how do we keep track of changes in these files upstream?

Reply:
Very good point. In this first draft I assembled the list of dependencies by hand directly inside the `flake.nix`, but it could just as well live in a separate file that gets imported using something like
```nix
bridle-deps = [
  # [...]
  (pkgs.python311.withPackages (ps: with ps; (import ./python-deps.nix {pkgs = pypkgs;})))
];
```
with `python-deps.nix` looking like
```nix
{pkgs}: [
  pkgs.dependency_1
  pkgs.dependency_2
  # [...]
]
```
This file could then fairly easily be auto-generated using a python script by parsing the `requirements.txt` we care about satisfying, similarly to how the workspace flake is generated based on west manifests. Some care would have to be taken to treat packages that aren't in `nixpkgs`, but that is only a handful of special cases.

I've already talked to @rexut about creating a west subcommand (e.g. `west nix --update`) to encapsulate re-generating such files, to make it as seamless as possible. If this is then run after `west update`, the flake should stay in sync with what's in the workspace.

Reply:
> I've already talked to @rexut about creating a west subcommand (e.g. west nix --update) to encapsulate re-generating such files, to make it as seamless as possible. If this is then run after west update, the flake should stay in sync with what's in the workspace.

That's a fabulous idea. If we already had that with an initial commit in this PR, the startup for newbies (like me) would probably be faster.